src/net.cpp

```diff
@@ -589,14 +589,20 @@ bool CNode::ReceiveMsgBytes(const char *pch, unsigned int nBytes, bool& complete
     while (nBytes > 0) {
         // absorb network data
         int handled = m_deserializer->Read(pch, nBytes);
-        if (handled < 0) return false;
+        if (handled < 0) {
+            return false;
+        }
 
         pch += handled;
         nBytes -= handled;
 
         if (m_deserializer->Complete()) {
             // decompose a transport agnostic CNetMessage from the deserializer
-            CNetMessage msg = m_deserializer->GetMessage(Params().MessageStart(), nTimeMicros);
+            Optional<CNetMessage> result = m_deserializer->GetMessage(Params().MessageStart(), nTimeMicros);
+            if (!result) {
+                continue;
+            }
+            CNetMessage msg = *result;
```

Comment:
in commit 5ad7ce11027536fb6e48098db9659744514a7fc2:

Why is this copy and alias of the message needed?

* `msg.` occurs only twice, so I think simply replacing `msg.` with `msg->` should be preferable to copying the whole message (or even adding a convenience alias (reference) to it).

Reply:
In my mind this was the safe way to use an optional (immediately check the optional and resolve it to a real object).  In this case though I've checked and what you've suggested both works and looks clear, so I'll go with it.